### PR TITLE
[codestyle] fix inefficient char to String conversion

### PR DESCRIPTION
**Pull request by CandyZQ**

### CODE REVIEW

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java [hunk 1]
```diff
@@ -80,12 +80,12 @@
      * Escapes to remove from the query string, anchor and interwiki parts when parsing the raw reference.
      */
     private static final String[] ESCAPES_EXTRA = new String[] { ESCAPE_CHAR + SEPARATOR_QUERYSTRING,
-        ESCAPE_CHAR + SEPARATOR_INTERWIKI, ESCAPE_CHAR + SEPARATOR_ANCHOR, "" + ESCAPE_CHAR + ESCAPE_CHAR };
+        ESCAPE_CHAR + SEPARATOR_INTERWIKI, ESCAPE_CHAR + SEPARATOR_ANCHOR, String.valueOf(ESCAPE_CHAR + ESCAPE_CHAR) };
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java]: This does not mean the same thing, you end up with a single chat with this code.

**Comment by CandyZQ** [comment]
Fixed

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java [hunk 2]
```diff
@@ -80,12 +80,12 @@
      * Escapes to remove from the query string, anchor and interwiki parts when parsing the raw reference.
      */
     private static final String[] ESCAPES_EXTRA = new String[] { ESCAPE_CHAR + SEPARATOR_QUERYSTRING,
-        ESCAPE_CHAR + SEPARATOR_INTERWIKI, ESCAPE_CHAR + SEPARATOR_ANCHOR, "" + ESCAPE_CHAR + ESCAPE_CHAR };
+        ESCAPE_CHAR + SEPARATOR_INTERWIKI, ESCAPE_CHAR + SEPARATOR_ANCHOR, String.valueOf(ESCAPE_CHAR )+ String.valueOf(ESCAPE_CHAR) };
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java]: The second String.valueOf is not needed.

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java [hunk 3]
```diff
@@ -80,12 +80,12 @@
...
 
     /**
      * Escapes to remove the interwiki content.
      */
-    private static final String[] ESCAPE_INTERWIKI = new String[] { "" + ESCAPE_CHAR + ESCAPE_CHAR, "" + ESCAPE_CHAR };
+    private static final String[] ESCAPE_INTERWIKI = new String[] { String.valueOf(ESCAPE_CHAR + ESCAPE_CHAR), String.valueOf(ESCAPE_CHAR) };
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java]: You missed this one.

**Comment by CandyZQ** [comment]
Sorry

**Comment by tmortagne** [comment]
Thanks for your contribution !

**Pull request merged**